sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -342,7 +347,10 @@ object JoinReorderDP extends PredicateHelper with Logging {
     /** Get the cost of the root node of this plan tree. */
     def rootCost(conf: SQLConf): Cost = {
       if (itemIds.size > 1) {
-        val rootStats = plan.stats
+        val rootStats = (plan transform {
+          // The OrderedJoin does not propagate stats
+          case OrderedJoin(left, right, jt, cond) => Join(left, right, jt, cond, JoinHint.NONE)
+        }).stats
```

Comment:
This is an unfortunate consequence of starting the reordering from the bottom. If anybody has suggestions on how to avoid this, I would happily hear you.

Reply:
This is outdated